src/Namotion.Reflection/Context/ContextualTypeExtensions.cs

```diff
@@ -261,6 +265,11 @@ public static ContextualType ToContextualType(this Type type)
         /// <returns>The <see cref="CachedType"/>.</returns>
         public static CachedType ToCachedType(this Type type)
         {
+            if (type.FullName == null)
```

Comment:
What strange case is this?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.type.fullname?view=net-5.0
> The fully qualified name of the type, including its namespace but not its assembly; or `null` if the current instance represents a generic type parameter, an array type, pointer type, or `byref` type based on a type parameter, or a generic type that is not a generic type definition but contains unresolved type parameters.

In my case, I'm inspecting the open generics, and with this line:
> https://github.com/RicoSuter/Namotion.Reflection/pull/60/files#diff-82e5a6a45918474d02929539d6aa5f4158e8b2f81f985f186b911c8971ae7baaR49

I get the real T type which has a `null` FullName, which obviously would yield the same cached results for every other T, because the key would be `Type:`. Didn't test that, but when I saw `null`, I was pretty sure that something bad would happen.

There might be other places where this should be checked though.

Reply:
Ah ok, thanks for the clarification